### ISSUE TITLE
Fix windows travis tests on master branch

### ISSUE THREAD

**Issue by joshrabinowitz** [sobolevn/git-secret#409]
In #372, we added support for testing on windows with travis-ci. 

After merging, however, the windows travis test is failing -- but only on windows on the master branch. 

The code  (in .ci/before_script.sh) passed tests before merge and passes tests in a PR, dev machine, and in docker with kitchen. 

See: https://travis-ci.org/sobolevn/git-secret/jobs/504169558 ,
where the log file ends with
````
$ if [[ "$TRAVIS_OS_NAME" != "windows" ]]; then gem update --system && gem install bundler; fi
$ chmod +x ".ci/before_script.sh" && ".ci/before_script.sh"
````

At https://travis-ci.org/sobolevn/git-secret/jobs/504088647
you can see the same code pass tests on windows

**Comment by joshrabinowitz** [comment]
@Mohjive @sobolevn @simbo1905 do you have any idea what's going wrong here?

**Comment by joshrabinowitz** [comment]
@sobolevn , the best process I can devise to fix this is to revert #410 and then debug and fix it directly in the master branch, which is sort of undesirable. 

Should I go ahead and do that, or can you think of a better process to fix this issue, or perhaps a third idea?

**Comment by simbo1905** [comment]
Well it looks like it passed a PR test and should work on master but it doesn’t. So while in general we shouldn’t break master here we have an odd problem that seems to exist on master. Windows testing on Travis-ci is early release, so it might be something odd in the infrastructure. So my vote is sure, break master and try to debug it. If you cannot crack it then fix master and then at can reflect on whether to park this until windows builds are production stable.

**Comment by simbo1905** [comment]
we have one open PR but I am fine with that staying in a holding pattern for this. we don’t want to break master for too long as it’s bad publicity for potential new contributors. what time limit should we set for trying to debug on master?

**Comment by joshrabinowitz** [comment]
I had a thought - maybe can I test this on master of _my private fork_? 
Then we wouldn't break the official master branch while we fix this.

**Comment by joshrabinowitz** [comment]
If we are going to do it on master, I'd put a time limit of (say) two hours before reverting to known good status.

**Comment by Mohjive** [comment]
Since the same code passed and failed on different occasions, my guess is that it is due to build environment not being stable. It also looked more as it timed out on `before_script.sh` (running `choco install gawk shellcheck`) rather than a code error.

**Comment by joshrabinowitz** [comment]
Yes it definitely appears to be failing on `before_script.sh` and not in a git-secret code error. 

So if we break the git-secret 'build-passing' badge for a while, it's sort of spurious, because the code base isn't broken, one surrounding script in the travis environment it.

I'll see if I can replicate the issue on the master branch of my git-secret fork.

**Comment by joshrabinowitz** [comment]
blocked by #416, which may be a temporary upstream issue (?)

**Comment by joshrabinowitz** [comment]
@simbo Are we using 'Travis CI encrypted environment variables'?

If so we're probably hitting this:
https://travis-ci.community/t/current-known-issues-please-read-this-before-posting-a-new-topic/264/9 which says:

>Issue :point_right: :point_right: If you have a secure variable, Windows will hang.
> In other words, a Travis CI encrypted environment variable in .travis.yml OR an environment variable in the Travis CI UI that doesn’t have “Display value in build log” checked, will cause a Windows job to hang.

**Comment by simbo1905** [comment]
I did ask @sobolevin to add an oauth token to travis so that we can automate the release. So I expect he did click the button to hide it in the logs it anyone can steal it.

**Comment by simbo1905** [comment]
Also their will be other keys to do deployments such as bintray. I am guessing their are many environment variables that are masked. 

Windows testing it beta I suggest we put it on hold until they fix this issue.

**Comment by joshrabinowitz** [comment]
This issue has been replaced by #756
